contracts/l1/L1DAITokenBridge.sol

```diff
@@ -0,0 +1,165 @@
+// Note: when bridge is closed it will still process in progress messages
+
+contract L1DAITokenBridge is iOVM_L1ERC20Bridge, OVM_CrossDomainEnabled {
+  // --- Auth ---
+  mapping (address => uint256) public wards;
+  function rely(address usr) external auth {
+    wards[usr] = 1;
+    emit Rely(usr);
+  }
+  function deny(address usr) external auth {
+    wards[usr] = 0;
+    emit Deny(usr);
+  }
+  modifier auth {
+    require(wards[msg.sender] == 1, "L1DAITokenBridge/not-authorized");
+    _;
+  }
+
+  event Rely(address indexed usr);
+  event Deny(address indexed usr);
+
+  address public immutable l1Token;
+  address public immutable l2DAITokenBridge;
+  address public immutable l2Token;
+  address public immutable escrow;
+  uint256 public isOpen = 1;
+
+  constructor(
+    address _l1Token,
+    address _l2Gateway,
+    address _l2Token,
+    address _l1messenger,
+    address _escrow
+  ) OVM_CrossDomainEnabled(_l1messenger) { 
+    wards[msg.sender] = 1;
+    emit Rely(msg.sender);
+
+    l1Token = _l1Token;
+    l2DAITokenBridge = _l2Gateway;
+    l2Token = _l2Token;
+    escrow = _escrow;
+  }
+
+  // --- Administration ---
+
+  function close() external auth {
+    isOpen = 0;
+  }
+
+  // --- Internal methods ---
+
+    function depositERC20(
+        address _l1Token,
+        address _l2Token,
+        uint256 _amount,
+        uint32 _l2Gas,
+        bytes calldata _data
+    )
+        external
+        override
+        virtual
+    {
+        require(_l1Token == l1Token && _l2Token == l2Token, "L1DAITokenBridge/token-not-dai");
+        
+        _initiateERC20Deposit(msg.sender, msg.sender, _amount, _l2Gas, _data);
+    }
+
+    function depositERC20To(
+        address _l1Token,
+        address _l2Token,
+        address _to,
+        uint256 _amount,
+        uint32 _l2Gas,
+        bytes calldata _data
+    )
+        external
+        override
+        virtual
+    {
+        require(_l1Token == l1Token && _l2Token == l2Token, "L1DAITokenBridge/token-not-dai");
+
+        _initiateERC20Deposit(msg.sender, _to, _amount, _l2Gas, _data);
+    }
+
+    function _initiateERC20Deposit(
+        address _from,
+        address _to,
+        uint256 _amount,
+        uint32 _l2Gas,
+        bytes calldata _data
+    )
+        internal
+    {
+        require(isOpen == 1, 'L1DAITokenBridge/closed');
+
+        TokenLike(l1Token).transferFrom(_from, escrow, _amount);
+
+        // Construct calldata for _l2Token.finalizeDeposit(_to, _amount)
+        bytes memory message = abi.encodeWithSelector(
+            iOVM_L2ERC20Bridge.finalizeDeposit.selector,
+            l1Token,
+            l2Token,
+            _from,
+            _to,
+            _amount,
+            _data
+        );
+
+        // Send calldata into L2
+        sendCrossDomainMessage(
+            l2DAITokenBridge,
+            _l2Gas,
+            message
+        );
+
+        // We omit _data here because events only support bytes32 types.
```

Comment:
@elenadimitrova is this comment still valid? I see that _data IS emitted.

Reply:
Yes, we're removing it in our PR. Feel free to delete